.github/workflows/test.yaml

```diff
@@ -40,30 +37,26 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
-    - name: Install poetry
-      run: pipx install poetry
-    - uses: actions/setup-python@v5
-      with:
-        python-version-file: pyproject.toml
-        cache: poetry
-    - name: Install dependencies
-      run: |
-        poetry install
+    - name: Install uv
+      uses: astral-sh/setup-uv@v3
+    - name: Setup python
+      run: uv python install
+    - name: Install the project
+      run: uv sync --all-extras --dev
```

Comment:
So we ran into this not too long ago in wristpy. Ruff doesn't actually need access to the dependencies right? So we could just install _only_ the dev dependencies (`--only-dev`).